realm/src/main/java/io/realm/android/RealmCursor.java

```diff
@@ -0,0 +1,776 @@
+     *   <li>{@link ColumnType#LINK}</li>
+     *   <li>{@link ColumnType#LINK_LIST}</li>
+     * </ul>
+     *
+     * <p>
+     * Some realm types are mapped to conform to the Cursor interface. These mappings are described below:
+     * <ol>
+     *   <li>boolean : Will return Cursor.FIELD_TYPE_INTEGER. {0 = false, 1 = true}, use getShort()/getInt()/getLong()</li>
+     *   <li>double : Will return Cursor.FIELD_TYPE_FLOAT, but must be fetched using use getDouble()</li>
+     *   <li>Date : Will return Cursor.FIELD_TYPE_INTEGER. Time in milliseconds since epoch, use getLong()</li>
+     *</ol>
+     *
+     * @param columnIndex Get the data type for this column index or -1 if the column isn't accessible.
+     * @return One of the {@code FIELD_TYPE_*}'s described in {@link Cursor}.
+     */
+    @Override
+    public int getType(int columnIndex) {
+        checkClosed();
+        ColumnType realmType = table.getColumnType(columnIndex);
+        switch (realmType) {
+            case BOOLEAN: return FIELD_TYPE_INTEGER;
+            case INTEGER: return FIELD_TYPE_INTEGER;
+            case FLOAT: return FIELD_TYPE_FLOAT;
+            case DOUBLE: return FIELD_TYPE_FLOAT;
+            case STRING: return FIELD_TYPE_STRING;
+            case BINARY: return FIELD_TYPE_BLOB;
+            case DATE: return FIELD_TYPE_INTEGER;
+            case TABLE:
+            case MIXED:
+            case LINK:
+            case LINK_LIST:
+            default:
+                return -1;
+        }
+    }
+
+    /**
+     * Realm currently doesn't support null. Calling this method will always throw an exception.
+     *
+     * @throws UnsupportedOperationException
+     */
+    @Override
+    public boolean isNull(int columnIndex) {
+        ColumnType realmType = table.getColumnType(columnIndex);
+        switch(realmType) {
+            case LINK: return table.isNullLink(columnIndex, rowIndex);
+            case LINK_LIST:
+            case BOOLEAN:
+            case INTEGER:
+            case FLOAT:
+            case DOUBLE:
+            case STRING:
+            case BINARY:
+            case DATE:
+            case TABLE:
+            case MIXED:
+            default:
+            throw new UnsupportedOperationException("isNull not yet supported by Realm for this type: " + realmType);
+        }
+    }
+
+    /**
+     * Realm doesn't support {@link #deactivate()}. Calling this will throw an exception.
+     *
+     * @throws UnsupportedOperationException
+     */
+    @Override
+    public void deactivate() {
+        throw new UnsupportedOperationException("As requery() is not supported, neither is deactivate()");
+    }
+
+    /**
+     * Realm doesn't support {@link #requery()}. Calling this will throw an exception.
+     *
+     * @throws UnsupportedOperationException
+     */
+    @Override
+    public boolean requery() {
+        throw new UnsupportedOperationException("Request a new cursor from the RealmResults or call Realm.refresh() instead");
```

Comment:
In case of a `ContentProvider`, this message could be ambiguous as the caller is not aware about the persistence layer backing this cursor. The generic message should be fine IMO